usage-collection/src/main/java/com/sequenceiq/cloudbreak/usage/MeteringEventProcessor.java

```diff
@@ -0,0 +1,74 @@
+import com.sequenceiq.cloudbreak.auth.crn.RegionAwareInternalCrnGeneratorFactory;
+import com.sequenceiq.cloudbreak.sigmadbus.model.DatabusRequest;
+import com.sequenceiq.cloudbreak.sigmadbus.model.DatabusRequestContext;
+import com.sequenceiq.cloudbreak.usage.metering.MeteringDatabusRecordProcessor;
+
+@Service
+public class MeteringEventProcessor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MeteringEventProcessor.class);
+
+    private static final int METERING_EVENT_VERSION = 3;
+
+    private final MeteringDatabusRecordProcessor meteringDatabusRecordProcessor;
+
+    private final RegionAwareInternalCrnGeneratorFactory internalCrnGeneratorFactory;
+
+    public MeteringEventProcessor(MeteringDatabusRecordProcessor meteringDatabusRecordProcessor,
+            RegionAwareInternalCrnGeneratorFactory internalCrnGeneratorFactory) {
+        this.meteringDatabusRecordProcessor = meteringDatabusRecordProcessor;
+        this.internalCrnGeneratorFactory = internalCrnGeneratorFactory;
+    }
+
+    public void processEvent(MeteringEventsProto.MeteringEvent event) {
+        DatabusRequestContext dbusContext = DatabusRequestContext.Builder.newBuilder()
+                .withAccountId(getAccountId())
+                .build();
+        DatabusRequest databusRequest = DatabusRequest.Builder.newBuilder()
+                .withMessageBody(event)
+                .withContext(dbusContext)
+                .build();
+        meteringDatabusRecordProcessor.processRecord(databusRequest);
+    }
+
+    public void storageHeartbeat(MeteringEventsProto.StorageHeartbeat details, MeteringEventsProto.ServiceType.Value serviceType) {
+        try {
+            checkNotNull(details);
+            checkNotNull(serviceType);
+            MeteringEventsProto.MeteringEvent event = eventBuilder()
+                    .setStorageHeartbeat(details)
+                    .setServiceType(serviceType)
+                    .setServiceConfiguration(MeteringEventsProto.ServiceConfiguration.newBuilder()
+                            .setServiceFeature(MeteringEventsProto.ServiceFeature.Value.OBJECT_STORAGE).build())
+                    .build();
+            processEvent(event);
+            LOGGER.info("Sent binary format for the following metering event: {}", details);
+        } catch (Exception e) {
+            LOGGER.warn("Could send binary format for the following metering event: {}! Cause: {}", details, e.getMessage());
+        }
+    }
+
+    private MeteringEventsProto.MeteringEvent.Builder eventBuilder() {
+        return MeteringEventsProto.MeteringEvent.newBuilder()
+                .setId(UUID.randomUUID().toString())
+                .setTimestamp(Instant.now().toEpochMilli())
+                .setVersion(METERING_EVENT_VERSION);
+    }
+
+    private String getAccountId() {
+        return Crn.safeFromString(internalCrnGeneratorFactory.iam().getInternalCrnForServiceAsString()).getAccountId();
+    }
```

Comment:
Sanity check: This is always `altus` (see [RegionAwareInternalCrnGeneratorUtil.INTERNAL_ACCOUNT](https://github.com/hortonworks/cloudbreak/blob/a23d7264b6579c4f89b70cc24e48b4ac8530bec8/common/src/main/java/com/sequenceiq/cloudbreak/auth/crn/RegionAwareInternalCrnGeneratorUtil.java#L7) ) . Perhaps you meant to retrieve the account ID from the current (real) user instead?

Reply:
I intentionally used the internal acc for the acc ID because it doesn't really matter what it is, DatabusRequestContext just needs to have the acc ID in it. We already use this approach in MessageBrokerUsageStrategy if the accountID is null or empty which is the case for nearly half of the usage events that we send.